Issue Definition designer Automation test

labkey-sweta opened the pull request:
#### Rationale
1. IssuesAdminTest.testCommentSortDirection
2. IssueAdminTest.testRelatedIssuesComments
3. IssueAdminTest.testProtectedFields
4. IssuesTest.defaultAssignedToTest

#### Related Pull Requests
* <!-- list of links to related pull requests (replace this comment) -->

#### Changes
* <!-- list of descriptions of changes that are worth noting (replace this comment) -->

labkey-sweta committed:
Issue Definition designer - new automated testing
1. IssuesAdminTest.testCommentSortDirection
2. IssueAdminTest.testRelatedIssuesComments
3. IssueAdminTest.testProtectedFields
4. IssuesTest.defaultAssignedToTest

labkey-tchad reviewed (changes requested):
The `newElementCache` methods should remain protected, not sure why they're getting switched to public here.

cnathe commented on src/org/labkey/test/tests/issues/IssuesAdminTest.java:
```diff
@@ -154,19 +167,102 @@ public void customIssueNameTest()
+        checker().verifyTrue("Title should be disabled", domainFormPanel.getField("Title").isFieldProtected());
+        checker().verifyTrue("NotifyList should be disabled", domainFormPanel.getField("NotifyList").isFieldProtected());
+        checker().verifyTrue("AssignedTo should be disabled", domainFormPanel.getField("AssignedTo").isFieldProtected());
+        checker().verifyTrue("Resolution should be disabled", domainFormPanel.getField("Resolution").isFieldProtected());
```
may be worth adding a couple of checks that some of the other default fields are not protected

labkey-sweta replied:
Done

cnathe commented on src/org/labkey/test/tests/issues/IssuesTest.java:
```diff
@@ -820,14 +819,28 @@ public void defaultAssignedToTest()
+        clickButton("Cancel");
 
-        // TODO: extend test to check validate full user selection list based on group selection...
         // TODO: compare user dropdown list between admin and new issues page
```
is this TODO still needed or is this covered in the check at line 843? maybe I am misreading it and there is more to do.

labkey-sweta replied:
I just read the second part or TODO worth adding  the test for "extend test to check validate full user selection list based on group selection" ?


> is this TODO still needed or is this covered in the check at line 843? maybe I am misreading it and there is more to do.

cnathe replied:
Got it.

cnathe reviewed (commented):
From a test case point-of-view, this looks to cover the items that we discussed. Thanks.

labkey-sweta committed:
Merge branch 'develop' into fb_IssuesAutomationTest

labkey-sweta committed:
Code review changes
1. Changing the public access modifier back to protected.
2. Adding the verification for non protected fields

labkey-tchad commented on src/org/labkey/test/components/domain/DomainFieldRow.java:
```diff
@@ -526,7 +536,8 @@ public DomainFieldRow setConceptLabelField(String labelField)
         expand();
-        elementCache().expectedVocabularyButton().click();;
+        elementCache().expectedVocabularyButton().click();
+        ;
```
```suggestion
        elementCache().expectedVocabularyButton().click();
```

labkey-sweta committed:
deleting the extra ;
